README.md
# CheckPoint SSL Hardering
Bu script Feridun ÖZTOK tarafından CheckPoint R81 ve R81.10 üzerinde varsayılan açık gelen zayıf şifreleme metotlarının devre dışı bırakılması için yazılmıştır. 

R81 için uygulanan makale <a href="https://supportcenter.checkpoint.com/supportcenter/portal?eventSubmit_doGoviewsolutiondetails=&solutionid=sk147272" target="_blank">sk147272</a>

Script ile düzenlenmeyen multi_portal_cipher_priority.conf dosyasını için expert modda cipher_util uygulamasını kullanmanız gerekmektedir. Rapid7 ile yapılan zafiyet taraması sonucu aşağıdaki metotları devre dışı bırakabilirsiniz.
İlgili döküman <a href="https://supportcenter.checkpoint.com/supportcenter/portal?eventSubmit_doGoviewsolutiondetails=&solutionid=sk126613" target="_blank">sk126613</a>


```
TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA
TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA
TLS_RSA_WITH_AES_128_CBC_SHA
TLS_RSA_WITH_AES_128_CBC_SHA256
TLS_RSA_WITH_AES_128_GCM_SHA256
TLS_RSA_WITH_AES_256_CBC_SHA
TLS_RSA_WITH_AES_256_CBC_SHA256
TLS_RSA_WITH_AES_256_GCM_SHA384
```

Sistem üzerinde özel tanım ya da HotFix olabileceğini göz önününde bulundurarak işlemleri gerçekleştirin. Elinizde mutlaka sağlam bir yedek olmasına dikkat edin.

İşlerin ters gitmesi ve yedeğinizin olmaması durumunda; expert ve CLI'da işletim sisteminize uygun çözümleri uygulamanız gerekmektedir.

# SSH geri alma yöntemi 1
```
rm /etc/ssh/templates/sshd_config.templ
cp /etc/ssh/templates/sshd_config.orj /etc/ssh/templates/sshd_config.templ
rm /etc/ssh/templates/sshd_config.orj
/bin/sshd_template_xlate < /config/active
service sshd reload 
```

# SSH geri alma yöntemi 2
```
vi /etc/ssh/templates/sshd_config.templ
KexAlgorithms +diffie-hellman-group14-sha1 ---> KexAlgorithms +diffie-hellman-group1-sha1
KexAlgorithms +diffie-hellman-group-exchange-sha256 ---> KexAlgorithms +diffie-hellman-group-exchange-sha1
/bin/sshd_template_xlate < /config/active
service sshd reload 
```

# R81 geri alma yöntemi 1

```
rm /web/templates/httpd-ssl.conf.templ
cp /web/templates/httpd-ssl.conf.orj /web/templates/httpd-ssl.conf.templ
rm /web/templates/httpd-ssl.conf.orj
```

# R81 geri alma yöntemi 2
```
chmod u+w /web/templates/httpd-ssl.conf.templ
vi /web/templates/httpd-ssl.conf.templ
SSLCipherSuite HIGH:!RC4:!LOW:!EXP:!aNULL:!SSLv2:!MD5 ---> SSLCipherSuite ECDHE-RSA-AES256-SHA384:AES256-SHA256:!ADH:!EXP:RSA:+HIGH:+MEDIUM:!MD5:!LOW:!NULL:!SSLv2:!eNULL:!aNULL:!RC4:!SHA1
SSLProtocol -ALL {ifcmp = $httpd:ssl3_enabled 1}+{else}-{endif}TLSv1.2 ---> SSLProtocol -ALL {ifcmp = $httpd:ssl3_enabled 1}+{else}-{endif}SSLv3 +TLSv1 +TLSv1.1 +TLSv1.2
chmod u-w /web/templates/httpd-ssl.conf.templ
```

# R81.10 geri alma yöntemi
```
set ssh server cipher 3des-cbc on
set ssh server cipher aes128-cbc on 
set ssh server cipher aes192-cbc on 
set ssh server cipher aes256-cbc on 
set ssh server cipher rijndael-cbc@lysator.liu.se on
set ssh server mac hmac-md5-96-etm@openssh.com on 
set ssh server mac hmac-md5-etm@openssh.com on 
set ssh server mac hmac-sha1 on 
set ssh server mac hmac-sha1-96-etm@openssh.com on 
set ssh server mac hmac-sha1-etm@openssh.com on 
set ssh server mac umac-64-etm@openssh.com on 
set ssh server mac umac-64@openssh.com on 
save config
```

Saygılarımla.


CHANGELOG.md
<body>
  <h1>Change Log</h1>

<p><h2>Sürüm V1.4 - 4 Ekim 2022</h2></p>
<p><h3>Eklenen</h3></p>
<p>R81 için SSH Cipher ve MACs değiştirme özelliği eklendi.</p>
<p><h3>Çıkarılan</h3></p>
<p>N/A</p>
<p><h3>Değiştirilen</h3></p>
<p>SSH servisi reload yerine restart edilmeye başlandı.</p>

<p><h2>Sürüm V1.3 - 28 Eylül 2022</h2></p>
<p><h3>Eklenen</h3></p>
<p>R81 için httpd-ssl.conf.templ dosyasını düzenleme özelliği eklendi.</p>
<p>Sürüm kontrol eklendi.</p>
<p>Sürüm güncelleme fonksiyonu eklendi. ./hardering.sh -u</p>
<p>Sürüm bilgisi öğrenme fonksiyonu eklendi. ./hardering.sh -v</p>
<p>Yardım menüsü eklendi. ./hardering.sh -h</p>
<p>Değiştirilen dosyaların işlem öncesi kopyası alınmaya başladı.</p>
<p><h3>Çıkarılan</h3></p>
<p>N/A</p>
<p><h3>Değiştirilen</h3></p>
<p>N/A</p>

<p><h2>Sürüm V1.2 - 27 Eylül 2022</h2></p>
<p><h3>Eklenen</h3></p>
<p>SSH server için "KexAlgorithms +diffie-hellman-group-exchange-sha1" "KexAlgorithms +diffie-hellman-group-exchange-sha256" ile değiştirilmesi için "sed -i 's/KexAlgorithms +diffie-hellman-group-exchange-sha1/KexAlgorithms +diffie-hellman-group-exchange-sha256/' /etc/ssh/templates/sshd_config.templ" komutu eklendi.</p>
<p><h3>Çıkarılan</h3></p>
<p>cipher_util için hazırlanan multi_portal_cipher_priority.conf dosya yapılandırması çalışmadığı için çıkartıldı. İlgili döküman numarası <a href="https://supportcenter.checkpoint.com/supportcenter/portal?eventSubmit_doGoviewsolutiondetails=&solutionid=sk126613" target="_blank">sk126613</a></p>
<p><h3>Değiştirilen</h3></p>
<p>"sed -i 's/KexAlgorithms +diffie-hellman-group1-sha1/#KexAlgorithms +diffie-hellman-group1-sha1/' /etc/ssh/templates/sshd_config.templ" komutu "sed -i 's/KexAlgorithms +diffie-hellman-group1-sha1/KexAlgorithms +diffie-hellman-group14-sha1/' /etc/ssh/templates/sshd_config.templ" ile değiştirildi.</p>

<p><h2>Sürüm V1.1 - 27 Eylül 2022</h2></p>
<p><h3>Eklenen</h3></p>
<p>Versiyon kontrol özellliği eklendi. Eğer versiyon R81.10 değilse script içindeki elemanlar görev yapmayacaktır.</p>
<p><h3>Çıkarılan</h3></p>
<p>N/A</p>
<p><h3>Değiştirilen</h3></p>
<p>N/A</p>

<p><h2>Sürüm V1.0 - 26 Eylül 2022</h2></p>
<p>Script'in ilk hali. Kodların çalıştığı test ortamında görüldü. Başka ortamlarda test edilmedi.</p>
<p><h3>Eklenen</h3></p>
<p>N/A</p>
<p><h3>Çıkarılan</h3></p>
<p>N/A</p>
<p><h3>Değiştirilen</h3></p>
<p>N/A</p>
</body>


hardering_v1.1.sh
#!/bin/bash
#====================================================================================================
# Baslik:           hardering.sh
# Kullanim:         ./hardering.sh
# Amaci:            CheckPoint R81.10 surumunda zayif sertifika tiplerini otomatik kapatma.
# Sahibi:           Feridun OZTOK
# Versiyon:         1.1
# Tarih:            27 Eylul 2022
#====================================================================================================
#
#====================================================================================================
#Degiskenler
source /etc/profile.d/CP.sh
#====================================================================================================
#
#====================================================================================================
#Cihaz kurulur kurulmaz bu script calistirilsa asagidaki dosyalarin sistem uzerinde olması germekte.
#Bu dosyalar expert uzerinden cli komutlarını cagirmak icin bize yardimci olacak.
file="/var/log/egis/g_bash"
if [ -f "$file" ]
then
 echo "$file found."
else
 echo 'HAtest="$2 $3 $4 $5 $6 $7 $8 $9"' > /var/log/egis/g_bash
 echo "echo \$HAtest > /var/log/g_command.txt"  >> /var/log/egis/g_bash
 echo "\$CPDIR/bin/cprid_util -server \$1 putfile -local_file /var/log/g_command.txt -remote_file /var/log/g_command.txt" >> /var/log/egis/g_bash
 echo "\$CPDIR/bin/cprid_util -server \$1 -verbose rexec -rcmd /bin/bash -f /var/log/g_command.txt" >> /var/log/egis/g_bash
 chmod 777 /var/log/egis/g_bash
fi
file="/var/log/egis/g_cli"
if [ -f "$file" ]
then
 echo "$file found."
else
 echo 'HAtest="$2 $3 $4 $5 $6 $7 $8 $9"' > /var/log/egis/g_cli
 echo "echo \$HAtest > /var/log/g_command.txt"  >> /var/log/egis/g_cli
 echo "\$CPDIR/bin/cprid_util -server \$1 putfile -local_file /var/log/g_command.txt -remote_file /var/log/g_command.txt" >> /var/log/egis/g_cli
 echo "\$CPDIR/bin/cprid_util -server \$1 -verbose rexec -rcmd /bin/clish -f /var/log/g_command.txt" >> /var/log/egis/g_cli
 chmod 777 /var/log/egis/g_cli
fi
file="/bin/c"
if [ -f "$file" ]
then
 echo "$file found."
else
 echo "echo \$@ > /var/log/clish.txt" > /bin/c
 echo "clish -f /var/log/clish.txt" >> /bin/c
 chmod 770 /bin/c
fi
#====================================================================================================
#
#====================================================================================================
#Database i serbes birakiyoruz.
clish -c "lock database override" >> /dev/null 2>&1
clish -c "lock database override" >> /dev/null 2>&1
#====================================================================================================
#
#====================================================================================================
#Ekran temizligi
clear
#====================================================================================================
#
#====================================================================================================
#Reklamlar
echo
echo
echo *#######################################################*
echo *#__________ CheckPoint SSL Hardering Script _________##*
echo *#____________________ Version 1.1 ___________________##*
echo *#_____________ Creator by Feridun OZTOK _____________##*
echo *#_ Egis Proje ve Danismanlik Bilisim Hiz. Ltd. Sti. _##*
echo *#____________ support@egisbilisim.com.tr ____________##*
echo *#######################################################*
echo
echo
#====================================================================================================
#
#====================================================================================================
#Sistem surum denetimi.
current_version=$(cat /etc/cp-release)
sistemsurum=""
if [[ $current_version == *"R81.10"* ]]; then
 echo "System Version  : R81.10 - Script calismaya uygun"
 sistemsurum="Dogru"
else
 echo "Script calismasi icin surum uygun degil!" 
fi
#====================================================================================================
#
#====================================================================================================
#Sistem tipi.
sistemtip=0
if [[ $($CPDIR/bin/cpprod_util FwIsFirewallModule 2> /dev/null) == *"1"*  ]]; then
   echo  "System Type     : System Type Gateway"
   sistemtip=1
elif [[ $($CPDIR/bin/cpprod_util FwIsFirewallMgmt 2> /dev/null) == *"1"*  ]]; then
   echo "System Type     : Management Server"
fi
#====================================================================================================
#
#====================================================================================================
#Sistem calisma suresi.
up_time=$(clish -c "show uptime")
echo "System Uptime   :"$up_time
#====================================================================================================
#
#====================================================================================================
if [[ $sistemsurum=="Dogru" ]]; then
echo "CLI tarafinda basit imzalar kapatiliyor."
clish -c "set ssh server cipher 3des-cbc off" 
clish -c "set ssh server cipher aes128-cbc off" 
clish -c "set ssh server cipher aes128-ctr on" 
clish -c "set ssh server cipher aes128-gcm@openssh.com on" 
clish -c "set ssh server cipher aes192-cbc off" 
clish -c "set ssh server cipher aes192-ctr on" 
clish -c "set ssh server cipher aes256-cbc off" 
clish -c "set ssh server cipher aes256-ctr on" 
clish -c "set ssh server cipher aes256-gcm@openssh.com on" 
clish -c "set ssh server cipher chacha20-poly1305@openssh.com on" 
clish -c "set ssh server cipher rijndael-cbc@lysator.liu.se off"
clish -c "set ssh server mac hmac-md5-96-etm@openssh.com off" 
clish -c "set ssh server mac hmac-md5-etm@openssh.com off" 
clish -c "set ssh server mac hmac-sha1 off" 
clish -c "set ssh server mac hmac-sha1-96-etm@openssh.com off" 
clish -c "set ssh server mac hmac-sha1-etm@openssh.com off" 
clish -c "set ssh server mac hmac-sha2-256 on" 
clish -c "set ssh server mac hmac-sha2-256-etm@openssh.com on" 
clish -c "set ssh server mac hmac-sha2-512 on" 
clish -c "set ssh server mac hmac-sha2-512-etm@openssh.com on" 
clish -c "set ssh server mac umac-64-etm@openssh.com off" 
clish -c "set ssh server mac umac-64@openssh.com off" 
clish -c "set ssh server mac umac-128-etm@openssh.com on" 
clish -c "set ssh server mac umac-128@openssh.com on" 
clish -c "save config"
fi
#====================================================================================================
#
#====================================================================================================
#Tekrar database i serbes birakiyoruz.
clish -c "lock database override" >> /dev/null 2>&1
clish -c "lock database override" >> /dev/null 2>&1
#====================================================================================================
#SSH tarafindaki diffie helman group 1 kapatiliyor.
echo "SSH tarafinda Diffie Hellman Group1 kapatiliyor."
sed -i 's/KexAlgorithms +diffie-hellman-group1-sha1/#KexAlgorithms +diffie-hellman-group1-sha1/' /etc/ssh/templates/sshd_config.templ
/bin/sshd_template_xlate < /config/active
service sshd reload
#====================================================================================================
#
#====================================================================================================
#Gateway icin basit imzalar devre disi birakiliyor.
if [ $sistemtip == 1 ]; then
echo "Multi Portaltarafinda basit imzalar kapatiliyor."
cd $FWDIR/conf
mv multi_portal_cipher_priority.conf multi_portal_cipher_priority.orj
echo "(" >> multi_portal_cipher_priority.conf
echo "        :version (3)" >> multi_portal_cipher_priority.conf
echo "        :tls_1_2_cipher_suites (" >> multi_portal_cipher_priority.conf
echo "                :list_name ("TLS 1.2 Ciphers")" >> multi_portal_cipher_priority.conf
echo "                :allowed (" >> multi_portal_cipher_priority.conf
echo "                        : (TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256)" >> multi_portal_cipher_priority.conf
echo "                        : (TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256)" >> multi_portal_cipher_priority.conf
echo "                        : (TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384)" >> multi_portal_cipher_priority.conf
echo "                        : (TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384)" >> multi_portal_cipher_priority.conf
echo "                        : (TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256)" >> multi_portal_cipher_priority.conf
echo "                        : (TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256)" >> multi_portal_cipher_priority.conf
echo "                        : (TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256)" >> multi_portal_cipher_priority.conf
echo "                        : (TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256)" >> multi_portal_cipher_priority.conf
echo "                        : (TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384)" >> multi_portal_cipher_priority.conf
echo "                        : (TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384)" >> multi_portal_cipher_priority.conf
echo "                        : (TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA)" >> multi_portal_cipher_priority.conf
echo "                        : (TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA)" >> multi_portal_cipher_priority.conf
echo "                        : (TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA)" >> multi_portal_cipher_priority.conf
echo "                        : (TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA)" >> multi_portal_cipher_priority.conf
echo "                        : (TLS_RSA_WITH_3DES_EDE_CBC_SHA)" >> multi_portal_cipher_priority.conf
echo "                        : (TLS_RSA_WITH_RC4_128_SHA)" >> multi_portal_cipher_priority.conf
echo "                        : (TLS_RSA_WITH_RC4_128_MD5)" >> multi_portal_cipher_priority.conf
echo "                )" >> multi_portal_cipher_priority.conf
echo "                :forbidden (" >> multi_portal_cipher_priority.conf
echo "                        : (TLS_RSA_WITH_AES_128_GCM_SHA256)" >> multi_portal_cipher_priority.conf
echo "                        : (TLS_RSA_WITH_AES_256_GCM_SHA384)" >> multi_portal_cipher_priority.conf
echo "                        : (TLS_RSA_WITH_AES_128_CBC_SHA256)" >> multi_portal_cipher_priority.conf
echo "                        : (TLS_RSA_WITH_AES_256_CBC_SHA256)" >> multi_portal_cipher_priority.conf
echo "                        : (TLS_RSA_WITH_AES_128_CBC_SHA)" >> multi_portal_cipher_priority.conf
echo "                        : (TLS_RSA_WITH_AES_256_CBC_SHA)" >> multi_portal_cipher_priority.conf
echo "                )" >> multi_portal_cipher_priority.conf
echo "        )" >> multi_portal_cipher_priority.conf
echo ")" >> multi_portal_cipher_priority.conf
chmod --reference=multi_portal_cipher_priority.orj multi_portal_cipher_priority.conf
fi
#====================================================================================================
#
#====================================================================================================
#Cikis.
exit
#====================================================================================================


hardering.sh
#!/bin/bash
#====================================================================================================
# Baslik:           hardering.sh
# Kullanim:         ./hardering.sh
# Amaci:            CheckPoint R81 ve R81.10 surumlerinde zayif sertifika tiplerini otomatik kapatma.
# Sahibi:           Feridun OZTOK
# Versiyon:         1.4
# Tarih:            4 Ekim 2022
#====================================================================================================
#Degiskenler
#====================================================================================================
source /etc/profile.d/CP.sh
MEVCUTSURUM="Script Versiyon  : 1.4"
#====================================================================================================
#show_version_info Fonksiyon
#====================================================================================================
show_version_info()
{
	echo ""
	echo "Script Versiyon  : 1.4"  
	echo "Script Tarihi    : 4 Ekim 2022"  
	echo "Son Guncelleyen  : Feridun OZTOK"
	echo ""
	exit 0
}
#====================================================================================================
#versiyon_kontrol Fonksiyon
#====================================================================================================
versiyon_kontrol()
{
	sed '2,$d' surumyakala.txt > surumazalt.txt
	awk -F"MEVCUTSURUM="  '{print $2}' surumazalt.txt > surumkisa.txt
	sed 's/"//g' surumkisa.txt > surumtemizlenmis.txt
	GUNCELSURUM=$(<surumtemizlenmis.txt)
	rm surum*
	if [[ "$MEVCUTSURUM" == "$GUNCELSURUM" ]]
		then
			echo "$MEVCUTSURUM" "Script calismaya uygun"
         else
			echo "Kullanilan surum guncel degil. Surumun $GUNCELSURUM olmasi gerekiyor."
         echo "./hardering.sh -u komutu ile guncelleyebilirsiniz. Script kapanacak."
         exit
	fi		
}
#====================================================================================================
#show_help_infoFonksiyon
#====================================================================================================
show_help_info()
{
   echo ""
   echo "Bu script, Feridun OZTOK tarafindan Rapid7 taramasi sonucu CheckPoint"
	echo "uzerinde cikan SSL - TLS versiyon aciklarinin giderilmesi icin yazilmistir."
   echo "https://github.com/feroooo/CheckPoint-SSL-Hardering/tree/master"
	echo ""
	echo "Script SMS ve Gateway uzerinde calisabilmektedir."
	echo ""
	echo "Script ./hardering.sh seklinde calisir. Kullanilabilir diger parametreler -v -u -h 'dir"
	echo ""
	echo "./hardering.sh -v ile mecvut scriptin surumunu ogrenebilirsiniz."
	echo "./hardering.sh -u ile script surumunu guncelleyebilirsiniz."
	echo "./hardering.sh -h ve diger tum tuslar su an okudugunuz yardim menusunu getirecektir."
	echo ""
	exit 0
}
#====================================================================================================
#download_updates Fonksiyon
#====================================================================================================
download_updates()
{
	rm hardering.sh
	curl_cli http://dynamic.egisbilisim.com.tr/script/hardering.sh | cat > hardering.sh && chmod 770 hardering.sh
	exit 0
}
#====================================================================================================
#Fonksiyon Tuslari
#====================================================================================================
while getopts ":v :u :h" opt; do
    case "${opt}" in
        h)
            show_help_info
            ;;
        u)
			   download_updates
            ;;
        v)
            show_version_info
            ;;
        *)
            #Catch all for any other flags
            show_help_info
            exit 1
            ;;
    esac
done
#====================================================================================================
#Dosya Kontrol
#====================================================================================================
#Cihaz kurulur kurulmaz bu script calistirilsa asagidaki dosyalarin sistem uzerinde olması germekte.
#Bu dosyalar expert uzerinden cli komutlarını cagirmak icin bize yardimci olacak.
file="/var/log/egis/g_bash"
if [ -f "$file" ]
then
 echo "$file found."
else
 echo 'HAtest="$2 $3 $4 $5 $6 $7 $8 $9"' > /var/log/egis/g_bash
 echo "echo \$HAtest > /var/log/g_command.txt"  >> /var/log/egis/g_bash
 echo "\$CPDIR/bin/cprid_util -server \$1 putfile -local_file /var/log/g_command.txt -remote_file /var/log/g_command.txt" >> /var/log/egis/g_bash
 echo "\$CPDIR/bin/cprid_util -server \$1 -verbose rexec -rcmd /bin/bash -f /var/log/g_command.txt" >> /var/log/egis/g_bash
 chmod 777 /var/log/egis/g_bash
fi
file="/var/log/egis/g_cli"
if [ -f "$file" ]
then
 echo "$file found."
else
 echo 'HAtest="$2 $3 $4 $5 $6 $7 $8 $9"' > /var/log/egis/g_cli
 echo "echo \$HAtest > /var/log/g_command.txt"  >> /var/log/egis/g_cli
 echo "\$CPDIR/bin/cprid_util -server \$1 putfile -local_file /var/log/g_command.txt -remote_file /var/log/g_command.txt" >> /var/log/egis/g_cli
 echo "\$CPDIR/bin/cprid_util -server \$1 -verbose rexec -rcmd /bin/clish -f /var/log/g_command.txt" >> /var/log/egis/g_cli
 chmod 777 /var/log/egis/g_cli
fi
file="/bin/c"
if [ -f "$file" ]
then
 echo "$file found."
else
 echo "echo \$@ > /var/log/clish.txt" > /bin/c
 echo "clish -f /var/log/clish.txt" >> /bin/c
 chmod 770 /bin/c
fi
#====================================================================================================
#Database unlock
#====================================================================================================
clish -c "lock database override" >> /dev/null 2>&1
clish -c "lock database override" >> /dev/null 2>&1
#====================================================================================================
#Ekran temizligi
#====================================================================================================
rm surum*
curl_cli http://dynamic.egisbilisim.com.tr/script/hardering.sh | grep "MEVCUTSURUM" > surumyakala.txt
clear
#====================================================================================================
#Reklamlar
#====================================================================================================
echo
echo
echo *#######################################################*
echo *#__________ CheckPoint SSL Hardering Script _________##*
echo *#____________________ Version 1.4 ___________________##*
echo *#_____________ Creator by Feridun OZTOK _____________##*
echo *#_ Egis Proje ve Danismanlik Bilisim Hiz. Ltd. Sti. _##*
echo *#____________ support@egisbilisim.com.tr ____________##*
echo *#######################################################*
echo
echo
#====================================================================================================
#Sistem surum denetimi
#====================================================================================================
current_version=$(cat /etc/cp-release)
sistemsurum=""
if [[ $current_version == *"R81.10"* ]]; then
	echo "Sistem Versiyon  : R81.10 - Script calismaya uygun"
	sistemsurum="CLI"
elif [[ $current_version == *"R81"* ]]; then
	echo "Sistem Versiyon  : R81 - Script calismaya uygun"
	sistemsurum="Klasik"
fi
#====================================================================================================
#Sistem tipi
#====================================================================================================
sistemtip=""
if [[ $($CPDIR/bin/cpprod_util FwIsFirewallModule 2> /dev/null) == *"1"*  ]]; then
   echo  "Sistem Tipi      : Security Gateway"
   sistemtip="Gateway"
elif [[ $($CPDIR/bin/cpprod_util FwIsFirewallMgmt 2> /dev/null) == *"1"*  ]]; then
   echo "Sistem Tipi      : Management Server"
   sistemtip="Management"
fi
#====================================================================================================
#Sistem calisma suresi
#====================================================================================================
up_time=$(clish -c "show uptime")
echo "Sistem Acik      :"$up_time
#====================================================================================================
#Script calistigi zaman versiyon kontrolu yap.
#====================================================================================================
versiyon_kontrol
#====================================================================================================
#R81.10 icin metot degisimi
#====================================================================================================
if [[ "$sistemsurum" == "CLI" ]]; then
	echo
	echo "CLI tarafinda metot degisim islemi basliyor."
	clish -c "set ssh server cipher 3des-cbc off" 
	echo "set ssh server cipher 3des-cbc off" 
	clish -c "set ssh server cipher aes128-cbc off" 
	echo "set ssh server cipher aes128-cbc off" 
	clish -c "set ssh server cipher aes128-ctr on" 
	echo "set ssh server cipher aes128-ctr on" 
	clish -c "set ssh server cipher aes128-gcm@openssh.com on" 
	echo "set ssh server cipher aes128-gcm@openssh.com on" 
	clish -c "set ssh server cipher aes192-cbc off" 
	echo "set ssh server cipher aes192-cbc off" 
	clish -c "set ssh server cipher aes192-ctr on" 
	echo "set ssh server cipher aes192-ctr on" 
	clish -c "set ssh server cipher aes256-cbc off" 
	echo "set ssh server cipher aes256-cbc off" 
	clish -c "set ssh server cipher aes256-ctr on" 
	echo "set ssh server cipher aes256-ctr on" 
	clish -c "set ssh server cipher aes256-gcm@openssh.com on" 
	echo "set ssh server cipher aes256-gcm@openssh.com on" 
	clish -c "set ssh server cipher chacha20-poly1305@openssh.com on" 
	echo "set ssh server cipher chacha20-poly1305@openssh.com on" 
	clish -c "set ssh server cipher rijndael-cbc@lysator.liu.se off"
	echo "set ssh server cipher rijndael-cbc@lysator.liu.se off"
	clish -c "set ssh server mac hmac-md5-96-etm@openssh.com off" 
	echo "set ssh server mac hmac-md5-96-etm@openssh.com off" 
	clish -c "set ssh server mac hmac-md5-etm@openssh.com off" 
	echo "set ssh server mac hmac-md5-etm@openssh.com off" 
	clish -c "set ssh server mac hmac-sha1 off" 
	echo "set ssh server mac hmac-sha1 off" 
	clish -c "set ssh server mac hmac-sha1-96-etm@openssh.com off" 
	echo "set ssh server mac hmac-sha1-96-etm@openssh.com off" 
	clish -c "set ssh server mac hmac-sha1-etm@openssh.com off" 
	echo "set ssh server mac hmac-sha1-etm@openssh.com off" 
	clish -c "set ssh server mac hmac-sha2-256 on" 
	echo "set ssh server mac hmac-sha2-256 on" 
	clish -c "set ssh server mac hmac-sha2-256-etm@openssh.com on" 
	echo "set ssh server mac hmac-sha2-256-etm@openssh.com on" 
	clish -c "set ssh server mac hmac-sha2-512 on" 
	echo "set ssh server mac hmac-sha2-512 on" 
	clish -c "set ssh server mac hmac-sha2-512-etm@openssh.com on" 
	echo "set ssh server mac hmac-sha2-512-etm@openssh.com on" 
	clish -c "set ssh server mac umac-64-etm@openssh.com off" 
	echo "set ssh server mac umac-64-etm@openssh.com off" 
	clish -c "set ssh server mac umac-64@openssh.com off" 
	echo "set ssh server mac umac-64@openssh.com off" 
	clish -c "set ssh server mac umac-128-etm@openssh.com on" 
	echo "set ssh server mac umac-128-etm@openssh.com on" 
	clish -c "set ssh server mac umac-128@openssh.com on" 
	echo "set ssh server mac umac-128@openssh.com on" 
	clish -c "save config"
	echo
	echo "CLI tarafinda metot degisim islemi bitti. Ayarlar kaydedildi."
	echo
fi
#====================================================================================================
#R81 icin metot degisimi
#====================================================================================================
if [[ "$sistemsurum" == "Klasik" ]]; then
	cp /web/templates/httpd-ssl.conf.templ /web/templates/httpd-ssl.conf.orj
	echo
	echo "httpd-ssl.conf.templ icin yazma yetkisi aliniyor."
	chmod u+w /web/templates/httpd-ssl.conf.templ
	echo "SSLCipherSuite metotlari degistiriliyor."
	sed -i 's/SSLCipherSuite HIGH:!RC4:!LOW:!EXP:!aNULL:!SSLv2:!MD5/SSLCipherSuite ECDHE-RSA-AES256-SHA384:AES256-SHA256:!ADH:!EXP:RSA:+HIGH:+MEDIUM:!MD5:!LOW:!NULL:!SSLv2:!eNULL:!aNULL:!RC4:!SHA1/' /web/templates/httpd-ssl.conf.templ
	echo "SSLv3 TLSv1 TLSv1.1 kapatiliyor."
	sed -i 's/SSLProtocol -ALL {ifcmp = $httpd:ssl3_enabled 1}+{else}-{endif}SSLv3 +TLSv1 +TLSv1.1 +TLSv1.2/SSLProtocol -ALL TLSv1.2/' /web/templates/httpd-ssl.conf.templ
	sed -i 's/SSLProtocol -ALL {ifcmp = $httpd:ssl3_enabled 1}+{else}-{endif}TLSv1.2/SSLProtocol -ALL TLSv1.2/' /web/templates/httpd-ssl.conf.templ
	echo "httpd-ssl.conf.templ icin yazma yetkisi kaldiriliyor."
	chmod u-w /web/templates/httpd-ssl.conf.templ
	/bin/template_xlate : /web/templates/httpd-ssl.conf.templ /web/conf/extra/httpd-ssl.conf < /config/active
	echo
	echo "httpd servisi yeniden baslatiliyor."
	tellpm process:httpd2
	tellpm process:httpd2 t
	echo
fi
#====================================================================================================
#SSH icin metot degisimi
#====================================================================================================
#Iki surum icinde /etc/ssh/templates/sshd_config.templ editlenecek.
cp /etc/ssh/templates/sshd_config.templ /etc/ssh/templates/sshd_config.orj
echo "SSH tarafindaki diffie helman group 1 group 14 ile degistiriliyor."
sed -i 's/KexAlgorithms +diffie-hellman-group1-sha1/KexAlgorithms +diffie-hellman-group14-sha1/' /etc/ssh/templates/sshd_config.templ
echo "SSH tarafindaki diffie helman group exchange-sha1 exchange-sha256 ile degistiriliyor."
sed -i 's/KexAlgorithms +diffie-hellman-group-exchange-sha1/KexAlgorithms +diffie-hellman-group-exchange-sha256/' /etc/ssh/templates/sshd_config.templ
echo "Ciphers +aes128-cbc kapatiliyor."
sed -i 's/Ciphers +aes128-cbc/#Ciphers +aes128-cbc/' /etc/ssh/templates/sshd_config.templ
echo
if [[ "$sistemsurum" == "Klasik" ]]; then
echo "Sistem R81.10 olmadigi icin CLI da yapilamayan degisiklik dosya uzerinde yapiliyor."
echo "aes128-cbc yerine chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com aciliyor."
sed -i '/\#Ciphers +aes128-cbc/a Ciphers chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com' /etc/ssh/templates/sshd_config.templ
echo "MACs tarafinda umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512 aktif ediliyor."
sed -i '/\Ciphers chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com/a MACs umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512' /etc/ssh/templates/sshd_config.templ
fi
/bin/sshd_template_xlate < /config/active
echo
echo "ssh servisi yeniden baslatiliyor."
service sshd restart
echo
#====================================================================================================
#Gateway icin cipher_util uyarisi
#====================================================================================================
if [[ "$sistemtip" == "Gateway" ]] ; then
	echo "Gateway uzerinde cipher_util ile ilave bir islem yapmaniz gerekmektedir. Asagida listelenen metotlari kapatmaniz tavsiye edilir."
	echo "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA"
	echo "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA"
	echo "TLS_RSA_WITH_AES_128_CBC_SHA"
	echo "TLS_RSA_WITH_AES_128_CBC_SHA256"
	echo "TLS_RSA_WITH_AES_128_GCM_SHA256"
	echo "TLS_RSA_WITH_AES_256_CBC_SHA"
	echo "TLS_RSA_WITH_AES_256_CBC_SHA256"
	echo "TLS_RSA_WITH_AES_256_GCM_SHA384"
	echo
	echo "cipher_util kullanimi icin: SK126613"
	echo
fi
#====================================================================================================
#Cikis
#====================================================================================================
exit

